### ISSUE TITLE
future data leakage

### ISSUE THREAD

**Issue by zhanghc12** [Ali-Meh619/SAMBA#6]
The normalizer was built using the entire dataset prior to the train/valid/test split. 

This approach resulted in data leakage, where information from the test set contaminated the training dataset.

That might explain why RankIC is high in the paper.

**Comment by Ali-Meh619** [comment]
Hello,
Thank you for your message.
The code came from different components and we put them together to make the github repository. I forgot to modify the pre-processing part for the final version, but we did the train/test split first and then applied the normalization, so there was no data leakage in the experiments. I will modify the code.
All the Best,
Ali
